src/dispatchers/pageDispatcher.ts

```diff
@@ -32,16 +32,19 @@ import { FileChooser } from '../server/fileChooser';
 import { CRCoverage } from '../server/chromium/crCoverage';
 import { JSHandle } from '../server/javascript';
 import { CallMetadata } from '../server/instrumentation';
+import { VideoDispatcher } from './videoDispatcher';
 
 export class PageDispatcher extends Dispatcher<Page, channels.PageInitializer> implements channels.PageChannel {
   private _page: Page;
 
-  constructor(scope: DispatcherScope, page: Page) {
+  // Note: Video must outlive Page and BrowserContext, so that client can saveAs it
+  // after closing the context.
+  constructor(scope: DispatcherScope, page: Page, scopeForVideo: DispatcherScope) {
     // TODO: theoretically, there could be more than one frame already.
     // If we split pageCreated and pageReady, there should be no main frame during pageCreated.
     super(scope, page, 'Page', {
       mainFrame: FrameDispatcher.from(scope, page.mainFrame()),
-      videoRelativePath: page._video ? page._video._relativePath : undefined,
+      video: page._video ? new VideoDispatcher(scopeForVideo, page._video) : undefined,
```

Comment:
If scope for video is on the parent level, the you should handle it on that level? Like Page has input, BrowserContext has video?